pymc/distributions/transforms.py

```diff
@@ -290,3 +294,62 @@ def bounds_fn(*rv_inputs):
 circular.__doc__ = """
 Instantiation of :class:`aeppl.transforms.CircularTransform`
 for use in the ``transform`` argument of a random variable."""
+
+
+class DiscreteRVTransform(RVTransform):
+    """Class of transforms that can be used for discrete variables"""
+
+    name = "discrete_transform"
+
+
+class DiscreteInterval(DiscreteRVTransform):
+    name = "dinterval"
+
+    def __init__(
+        self, args_fn: Callable[..., Tuple[Optional[TensorVariable], Optional[TensorVariable]]]
+    ):
+        """
+
+        Parameters
+        ==========
+        args_fn
+            Function that expects inputs of RandomVariable and returns the lower
+            and upper bounds for the modulo transformation. If one of these is
+            None, the RV is considered to be unbounded on the respective edge.
+        """
+        self.args_fn = args_fn
+
+    def forward(self, value: TensorVariable, *inputs: Variable) -> TensorVariable:
+        return value
+
+    def backward(self, value: TensorVariable, *inputs: Variable) -> TensorVariable:
+        lower, upper = self.args_fn(*inputs)
+
+        if lower is not None and upper is not None:
+            # This is probably less than ideal, as it will simply wrap around values
+            # between lower and upper, but most bounded distributions are likely to
+            # have very different masses near the two bounds.
+            return lower + at.mod((value - lower), (upper - lower + 1))
```

Comment:
What about a reflection? Something like this 

    at.switch((((value-lower) // (upper-lower+1)) % 2), 
            upper - np.mod(value - lower, upper - lower +1),
            lower + np.mod(value - lower, upper - lower +1))

Reply:
I can't quite decipher what the switch condition is doing, but reflection sounds much more wise